tests/unit/test_plotting/test_plot_voltage_components.py

```diff
@@ -7,77 +11,37 @@
 use("Agg")
 
 
-class TestPlotVoltageComponents:
-    def test_plot_with_solution(self):
-        model = pybamm.lithium_ion.SPM()
-        sim = pybamm.Simulation(model)
-        sol = sim.solve([0, 3600])
-        for split in [True, False]:
-            _, ax = pybamm.plot_voltage_components(
-                sol, show_plot=False, split_by_electrode=split
-            )
-            t, V = ax.get_lines()[0].get_data()
-            np.testing.assert_array_equal(t, sol["Time [h]"].data)
-            np.testing.assert_array_equal(V, sol["Battery voltage [V]"].data)
-
-        _, ax = plt.subplots()
-        _, ax_out = pybamm.plot_voltage_components(sol, ax=ax, show_legend=True)
-        assert ax_out == ax
-
-    def test_plot_with_simulation(self):
-        model = pybamm.lithium_ion.SPM()
-        sim = pybamm.Simulation(model)
-        sim.solve([0, 3600])
-
-        for split in [True, False]:
-            _, ax = pybamm.plot_voltage_components(
-                sim, show_plot=False, split_by_electrode=split
-            )
-            t, V = ax.get_lines()[0].get_data()
-            np.testing.assert_array_equal(t, sim.solution["Time [h]"].data)
-            np.testing.assert_array_equal(V, sim.solution["Battery voltage [V]"].data)
-
-        _, ax = plt.subplots()
-        _, ax_out = pybamm.plot_voltage_components(sim, ax=ax, show_legend=True)
-        assert ax_out == ax
+@pytest.fixture
+def solved_simulation():
+    model = pybamm.lithium_ion.SPM()
+    sim = pybamm.Simulation(model)
+    sol = sim.solve([0, 3600])
+    return sim, sol
 
-    def test_plot_from_solution(self):
-        model = pybamm.lithium_ion.SPM()
-        sim = pybamm.Simulation(model)
-        sol = sim.solve([0, 3600])
-        for split in [True, False]:
-            _, ax = sol.plot_voltage_components(
-                show_plot=False, split_by_electrode=split
-            )
-            t, V = ax.get_lines()[0].get_data()
-            np.testing.assert_array_equal(t, sol["Time [h]"].data)
-            np.testing.assert_array_equal(V, sol["Battery voltage [V]"].data)
 
-        _, ax = plt.subplots()
-        _, ax_out = sol.plot_voltage_components(ax=ax, show_legend=True)
-        assert ax_out == ax
+@pytest.mark.parametrize(
+    "from_solution", [True, False], ids=["from_solution", "from_simulation"]
+)
```

Comment:
pre-commit is not happy when you want to keep it in a single line?

Reply:
No it changes it to multi-line.

Reply:
Yes got changed to multi line in pre-commit